runtime/System/Console/Hawk/Runtime.hs

```diff
@@ -74,10 +74,14 @@ splitAtSeparator (Delimiter d) = Search.split d
 
 
 outputRows :: Rows a => OutputSpec -> a -> IO ()
-outputRows (OutputSpec _ spec) x = ignoringBrokenPipe $ do
+outputRows (OutputSpec out spec) x = ignoringBrokenPipe $ do
     let s = join' (toRows x)
-    B.putStr s
-    hFlush stdout
+    case out of
+        UseStdout                             -> B.putStr s >> hFlush stdout
+        OutputFile (FileSink f Nothing)       -> B.writeFile (f <.> tmp) s
+        OutputFile (FileSink f (Just backup)) -> do
+            copyFile f backup
+            B.writeFile (f <.> tmp) s
```

Comment:
I believe there is a function somewhere, probably in `filepath` or `posix`, for creating a temporary file in a location which is appropriate for the OS, such as `/tmp`. It would be better to create a file there, and to delete it afterwards using `bracket`, than to create a `.tmp` file in the current directory.